hack/cleanup_gke_clusters.go

```diff
@@ -136,22 +114,28 @@ func findOrphanedClusters(ctx context.Context, mgrc *container.ClusterManagerCli
 
 	var orphanedClusterNames []string
 	for _, cluster := range clusterListResp.Clusters {
-		if createdBy, ok := cluster.ResourceLabels[gke.GKECreateLabel]; ok {
-			if gcloudClientID == createdBy {
-				createdAt, err := time.Parse(time.RFC3339, cluster.CreateTime)
-				if err != nil {
-					return nil, err
-				}
-
-				orphanTime := createdAt.Add(timeUntilClusterOrphaned)
-				if time.Now().UTC().After(orphanTime) {
-					orphanedClusterNames = append(orphanedClusterNames, cluster.Name)
-				} else {
-					fmt.Printf("INFO: cluster %s skipped (built in the last %s)\n", cluster.Name, timeUntilClusterOrphaned)
-				}
+		if isCreatedByTests(cluster) {
+			createdAt, err := time.Parse(time.RFC3339, cluster.CreateTime)
+			if err != nil {
+				return nil, err
+			}
+
+			orphanTime := createdAt.Add(timeUntilClusterOrphaned)
+			if time.Now().UTC().After(orphanTime) {
+				orphanedClusterNames = append(orphanedClusterNames, cluster.Name)
+			} else {
+				fmt.Printf("INFO: cluster %s skipped (built in the last %s)\n", cluster.Name, timeUntilClusterOrphaned)
 			}
 		}
 	}
 
 	return orphanedClusterNames, nil
 }
+
+func isCreatedByTests(cluster *containerpb.Cluster) bool {
+	if cluster == nil {
+		return false
+	}
+
+	return e2e.IsCreatedByE2ETests(cluster.Name)
```

Comment:
Perhaps adding a label check (and adding that label on creation) would be a nice touch and more future proof but this should do for now.

Reply:
That's a good idea, let's unblock it with this PR: https://github.com/Kong/kubernetes-testing-framework/pull/499